typespec-extension/src/code-model-builder.ts

```diff
@@ -542,7 +542,7 @@ export class CodeModelBuilder {
       // server
       let baseUri = "{endpoint}";
       const servers = getServers(this.program, client.service);
-      if (servers && servers.length === 1) {
+      if (servers && servers.length === 1 && !this.isArmSynthesizedServer(servers[0])) {
         baseUri = servers[0].url;
       }
```

Comment:
This part was a bit strange (why code conditioned on `servers.length`). Thought? (I know I wrote this...)

Reply:
OK, probably it would only has 1 server anyway

Reply:
Yeah, we seem to deal with one server only: https://github.com/Azure/autorest.java/blob/95268e17df0d1728e1db71fed63e9a248d2c0df3/typespec-extension/src/code-model-builder.ts#L548